tao_compiler/mlir/disc/transforms/lhlo_elemental_utils.cc

```diff
@@ -959,6 +960,16 @@ Value elementalLower<lmhlo::CopyOp>(OpBuilder* b, Location loc,
   return result;
 }
 
+template <>
+Value elementalLower<lmhlo_disc::H2DOp>(OpBuilder* b, Location loc,
+                                        lmhlo_disc::H2DOp op,
+                                        ValueRange output_index,
+                                        bool check_cache,
+                                        LowerConfig* lower_config) {
+  Value zero = b->create<arith::ConstantIndexOp>(loc, 0);
+  return b->create<memref::LoadOp>(loc, op->getOperand(0), zero);
```

Comment:
Does that mean we will use uniform memory access, and direct access to CPU data from the device?

Reply:
yes, `gpu-kernel-outlining` will move single element CPU buffer to kernel parameter.